contrib/internal/httptrace/response_writer.go

```diff
@@ -16,6 +16,13 @@ type responseWriter struct {
 	status int
 }
 
+// ResetStatusCode resets the status code of the response writer.
+func ResetStatusCode(w http.ResponseWriter) {
+	if rw, ok := w.(*responseWriter); ok {
+		rw.status = 0
+	}
+}
+
```

Comment:
This required because of [this `if` statement](https://github.com/DataDog/dd-trace-go/blob/16269d5f15b910032c6372ce075cb9b7abe898f9/contrib/internal/httptrace/response_writer.go#L41-L43).